cvereporter/report.py

```diff
@@ -0,0 +1,50 @@
+from cyclonedx.exception import MissingOptionalDependencyException
+from cyclonedx.factory.license import LicenseFactory
+from cyclonedx.model import OrganizationalEntity, XsUri
+from cyclonedx.model.bom import Bom
+from cyclonedx.model.component import Component, ComponentType
+from cyclonedx.model.impact_analysis import ImpactAnalysisAffectedStatus
+from cyclonedx.model.vulnerability import Vulnerability, VulnerabilitySource,VulnerabilityScoreSource, VulnerabilityRating, VulnerabilitySeverity, BomTarget, BomTargetVersionRange
+from cyclonedx.output import make_outputter, LATEST_SUPPORTED_SCHEMA_VERSION
+from cyclonedx.output.json import JsonV1Dot4
+from cyclonedx.schema import SchemaVersion, OutputFormat
+from cyclonedx.validation.json import JsonStrictValidator
+from cyclonedx.validation import make_schemabased_validator
+from datetime import datetime
+
+# based on sample code from https://cyclonedx-python-library.readthedocs.io/en/latest/examples.html
+lc_factory = LicenseFactory()
+bom = Bom()
+bom.metadata.component = root_component = Component(
+    name='Eclipse Temurin',
+    type=ComponentType.APPLICATION,
+    licenses=[lc_factory.make_from_string('GPL v2')],
+    bom_ref='temurin-vdr',
+)
+
+
+vuln1 = Vulnerability(
+    id="CVE-2-23-25193",
+    source=VulnerabilitySource(name="NVD", url="https://nvd.nist.gov/vuln/detail/CVE-2023-25193"),
+    published=datetime.strptime("2023-02-04T20:15:08.027", "%Y-%m-%dT%H:%M:%S.%f"),
+    updated=datetime.strptime("2023-07-25T15:15:13.163", "%Y-%m-%dT%H:%M:%S.%f"),
+    description="hb-ot-layout-gsubgpos.hh in HarfBuzz through 6.0.0 allows attackers to trigger O(n^2) growth via consecutive marks during the process of looking back for base glyphs when attaching marks.",
+    recommendation="Upgrade to the latest version of Eclipse Temurin."
+)
+
+rating1 = VulnerabilityRating(
+    source=VulnerabilitySource(url= "https://openjdk.org/groups/vulnerability/advisories", name="OJVG"),
+    score=3.7,
+    severity=VulnerabilitySeverity.LOW,
+    method=VulnerabilityScoreSource.CVSS_V3_1,
+    vector="CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
+)
+vuln1.ratings.add(rating1)
```

Comment:
I assume we can add a 'rating2' for the NIST rating of this cve also?

Reply:
this was just an experiment, i will add nist ratings on the pr #2